pkg/apis/acme/v1/types_issuer.go

```diff
@@ -204,6 +204,14 @@ type ACMEChallengeSolverHTTP01 struct {
 	// provisioned by cert-manager for each Challenge to be completed.
 	// +optional
 	Ingress *ACMEChallengeSolverHTTP01Ingress `json:"ingress,omitempty"`
+
+	// The Istio virtualservice based HTTP01 challenge solver will solve
+	// challenges by creating an Istio virtualservice resource that is connected
+	// to the specified Istio gateway in order to route requests for
+	// '/.well-known/acme-challenge/XYZ' to 'challenge solver' pods that are
+	// provisioned by cert-manager for each Challenge to be completed.
+	// +optional
+	Istio *ACMEChallengeSolverHTTP01Istio `json:"istio,omitempty"`
```

Comment:
There is definitely a use case for modifying existing VirtualServices.
Because of the way Istio orders the rules across different VirtualServices, there could be a conflict with the cert-manager rule. This will be the case if a non-catch all rule is used (checkout https://istio.io/latest/docs/ops/best-practices/traffic-management/#split-virtual-services for more info).
Although, to update an excising VirtualServices, an "official" Istio binding should probably be used instead of the DeepCopy & wrapper code that is now used in this PR (blocked by istio/api#1959).
In my opinion adding logic for modifying VirualServices is out-of-scope for this PR.

Reply:
Ok. I agree that this can be added in a later PR.

Reply:
See #3950.

Reply:
About @wallrj 's comment above,

```go
// The ingress based HTTP01 challenge solver will solve challenges by
// creating or modifying Ingress resources in order to route requests for
// '/.well-known/acme-challenge/XYZ' to 'challenge solver' pods that are
// provisioned by cert-manager for each Challenge to be completed.
```

AFAIK "modify" means that we reuse the Ingress that was already created for solving the challenge, but it does not imply that cert-manager modifies the user-created Ingresses. I think we should change this comment to reflect that.

Reply:
> AFAIK "modify" means that we reuse the Ingress that was already created for solving the challenge, but it does not imply that cert-manager modifies the user-created Ingresses.

Actually it _does_ mean we modify the user-created Ingress. For ingress controllers like `ingress-gce` which do not 'flatten' Ingresses into a single serving endpoint, we support "edit in place" mode, which is what this refers to :)

Reply:
Aaah yes, I did not know about the [`edit-in-place`](https://cert-manager.io/docs/usage/ingress/#supported-annotations) annotation! Thanks for pointing it out.

```yaml
acme.cert-manager.io/http01-edit-in-place: "true"
```